File: src/my_robot_controller/my_robot_controller/add_two_client_2.py
from functools import partial

import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts


class AddTwoIntsClientNode(Node):
    def __init__(self):
        super().__init__('add_two_client2')
        self.get_logger().info('Add two int client has been started!')
        self.call_add_two_ints_server(1, 5)
        self.call_add_two_ints_server(2, 6)
        self.call_add_two_ints_server(3, 7)


    def call_add_two_ints_server(self, a, b):
        client = self.create_client(AddTwoInts, "add_two_ints")
        while not client.wait_for_service(1.0):
            self.get_logger().warn('Waiting for Server Add Two Ints...')

        request = AddTwoInts.Request()
        request.a = a
        request.b = b

        future = client.call_async(request)
        future.add_done_callback(
            partial(self.callback_call_add_two_ints, a=a, b=b))

    def callback_call_add_two_ints(self, future, a, b):
        try:
            response = future.result()
            self.get_logger().info(f"{a} + {b} = {response.sum}")
        except Exception as e:
            self.get_logger().error(f"Service call failed: {e}")


def main(args=None):
    rclpy.init(args=args)
    node = AddTwoIntsClientNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/add_two_client_2.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

class AddTwoClient : public rclcpp::Node {
public:
    AddTwoClient() : Node("add_two_client2") {
        threads_.push_back(std::thread(std::bind(&AddTwoClient::callAddTwoService, this, 1, 4)));
        threads_.push_back(std::thread(std::bind(&AddTwoClient::callAddTwoService, this, 2, 5)));
        threads_.push_back(std::thread(std::bind(&AddTwoClient::callAddTwoService, this, 3, 6)));
    }

    void callAddTwoService(int a, int b) {
        auto client = this->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints");
        while (!client->wait_for_service(std::chrono::seconds(1))) {
            RCLCPP_WARN(this->get_logger(), "Waiting for the server to be up...");
        }
        auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
        request->a = a;
        request->b = b;

        auto future = client->async_send_request(request);
        try {
            auto response = future.get();
            RCLCPP_INFO(this->get_logger(), "%d + %d = %d",
                a, b, static_cast<int>(response->sum));
        }
        catch (const std::exception& e) {
            RCLCPP_ERROR(this->get_logger(), "Error while calling the service!");
        }
    }

private:
    std::vector<std::thread> threads_;  // Create a thread pool
};

int main(int argc, char const* argv[]) {
    rclcpp::init(argc, argv);
    auto node = std::make_shared<AddTwoClient>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
